Sources/SwiftProtobuf/GetPathDecoder.swift

```diff
@@ -0,0 +1,325 @@
+// Sources/SwiftProtobuf/GetPathDecoder.swift - Path decoder (Getter)
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Decoder which captures value of a field by its path.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+struct GetPathDecoder<T: Message>: Decoder {
+
+  private let path: String
+  private var number: Int?
+
+  private var _value: Any?
+  private var _hasPath: Bool = false
```

Comment:
A value could be optional (and possibly nil) but its path exists.

Reply:
Oh right, since this is a `Decoder` we're feeding path segments to walk to the right place?  Probably should be a comment on the on generally how this decoder works for someone looking in the future.

Reply:
Can you add that mentioned comment?